samtranslator/model/api/api_generator.py

```diff
@@ -400,7 +399,7 @@ def _construct_deployment(self, rest_api: ApiGatewayRestApi) -> ApiGatewayDeploy
             self.logical_id + "Deployment", attributes=self.passthrough_resource_attributes
         )
         deployment.RestApiId = rest_api.get_runtime_attr("rest_api_id")
-        if not self.remove_extra_stage:
```

Comment:
Mind want to add a comment here to explain why we add a stage when open api version is not specified. https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-api.html#sam-api-openapiversion

Reply:
I'm not sure why we made this change because it's 5 years ago. But I will look into that and add later.